src/DotNetLightning.Core/Channel/Channel.fs

```diff
@@ -388,6 +388,32 @@ module Channel =
                 [] |> Ok
 
         // ---------- normal operation ---------
+        | ChannelState.Normal state, MonoHopUnidirectionalPayment cmd when state.LocalShutdown.IsSome || state.RemoteShutdown.IsSome ->
+            sprintf "Could not send mono-hop unidirectional payment %A since shutdown is already in progress." cmd
+            |> apiMisuse
+        | ChannelState.Normal state, MonoHopUnidirectionalPayment cmd ->
+            result {
+                let payment: MonoHopUnidirectionalPayment = {
+                    ChannelId = state.Commitments.ChannelId
+                    Amount = cmd.Amount
+                }
+                let commitments1 = state.Commitments.AddLocalProposal(payment)
+
+                let remoteCommit1 =
+                    match commitments1.RemoteNextCommitInfo with
+                    | RemoteNextCommitInfo.Waiting info -> info.NextRemoteCommit
+                    | RemoteNextCommitInfo.Revoked _info -> commitments1.RemoteCommit
```

Comment:
as revokation is not implemented yet, this line above is not needed right?

Reply:
We still need to use the correct remote commitment though. So changing this to always use `commitments1.RemoteCommit` would break it.

Reply:
sorry I don't understand

Reply:
Revocation isn't implemented in the sense that we don't validate or save the revocation secrets that the peer sends us. We still send and receive `revoke_and_ack` messages though as part of the process of updating the commitments.

Reply:
why do we need to send those messages if we don't do revokation?

Reply:
The protocol specifies that a peer must always reply with a `revoke_and_ack` when it receives a `commitment_signed`. DNL uses the receiving of the `revoke_and_ack` to progress to the next commitments state. It would have been harder (ie. required more changes to DNL) to remove the revoke_and_ack, and would have been a step backwards since we'll need it eventually anyway for when we add proper revocation handling.